tfx/components/example_gen/base_example_gen_executor_test.py

```diff
@@ -151,6 +155,29 @@ def testDoOutputSplit(self):
         tf.io.gfile.GFile(self._train_output_file).size(),
         tf.io.gfile.GFile(self._eval_output_file).size())
 
+  def testDoOutputSplitWithSequenceExample(self):
+    # Add output config to exec proterties.
+    self._exec_properties[utils.OUTPUT_CONFIG_KEY] = json_format.MessageToJson(
+        example_gen_pb2.Output(
+            split_config=example_gen_pb2.SplitConfig(splits=[
+                example_gen_pb2.SplitConfig.Split(name='train', hash_buckets=2),
+                example_gen_pb2.SplitConfig.Split(name='eval', hash_buckets=1)
+            ])))
+    self._exec_properties['sequence_example'] = True
+
+    # Run executor.
+    example_gen = TestExampleGenExecutor()
+    example_gen.Do({}, self._output_dict, self._exec_properties)
+
+    # Check example gen outputs.
+    self.assertTrue(tf.io.gfile.exists(self._train_output_file))
+    self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+
+    # Output split ratio: train:eval=2:1.
+    self.assertGreater(
+        tf.io.gfile.GFile(self._train_output_file).size(),
+        tf.io.gfile.GFile(self._eval_output_file).size())
```

Comment:
I noticed there are a lot of duplication for line 169-179 in this file, could you help us refactor it? (create a _testDo like importexamplegen executor_test)

Reply:
yes i see, i added _testDo